File: Algorithm/Strings/HackerRank in a String!/HackerRank in a String!.c
#include <math.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <assert.h>
#include <limits.h>
#include <stdbool.h>

char* hackerrankInString(char* s) {
    // Complete this function
    char* key = "hackerrank";
    int j = 0;
    for( int i = 0; i < strlen(s); i++){
        if( j < 10 && s[i] == key[j])
            j++;
    }
    if( j == 10)
        return "YES";
    else
        return "NO";
}

int main() {
    int q; 
    scanf("%i", &q);
    for(int a0 = 0; a0 < q; a0++){
        char* s = (char *)malloc(512000 * sizeof(char));
        scanf("%s", s);
        int result_size;
        char* result = hackerrankInString(s);
        printf("%s\n", result);
    }
    return 0;
}

File: Algorithm/Strings/HackerRank in a String!/HackerRank in a String!.py
#!/bin/python3

import sys

def hackerrankInString(s):
    # Complete this function
    key = "hackerrank"
    i = 0
    for char in s:
        if( i < 10 and char.lower() == key[i]):
            i += 1
    return "YES" if( i == 10) else "NO"

if __name__ == "__main__":
    q = int(input().strip())
    for a0 in range(q):
        s = input().strip()
        result = hackerrankInString(s)
        print(result)